Source/WebGPU/WGSL/TypeCheck.cpp

```diff
@@ -318,9 +318,35 @@ void TypeChecker::visit(AST::IdentifierExpression& identifier)
 
 void TypeChecker::visit(AST::CallExpression& call)
 {
-    auto* target = resolve(call.target());
-    // FIXME: validate arguments
-    inferred(target);
+    Vector<Type*> arguments;
+    arguments.reserveInitialCapacity(call.arguments().size());
+    for (auto& argument : call.arguments())
+        arguments.append(infer(argument));
+
+    auto& target = call.target();
+    if (is<AST::NamedTypeName>(target)) {
+        auto& namedTarget = downcast<AST::NamedTypeName>(target);
+        auto* result = chooseOverload(namedTarget.name(), arguments);
+        if (result)
+            inferred(result);
+        else {
+            StringPrintStream argumentsString;
```

Comment:
Not StringBuilder?

I'm consistently bewildered at the different in coding conventions between JSC and WebCore

Reply:
We seem to have both in WebGPU, I just opened a random file and used the same things that was being used 🤷‍♂️